fortuna/prob_model/posterior/normalizing_flow/advi/advi_posterior.py

```diff
@@ -74,11 +79,71 @@ def fit(
                 "`save_checkpoint_dir` must be passed when `dump_state` is set to True."
             )
 
-        init_prob_model_state, n_train_data, n_val_data = self._init(
-            train_data_loader, val_data_loader
+        state = None
+        stds = None
+        allowed_states = [MAPState, ADVIState]
+        status = dict()
+
+        if fit_config.checkpointer.restore_checkpoint_path is not None:
```

Comment:
What do you think about extracting a method for this part? It would improve readability

Reply:
Yes, I'll do that.